### PR TITLE
Visitor can choose articles by categories

### PR DESCRIPTION
**Pull request by FacundoOs**
[PT](https://www.pivotaltracker.com/story/show/175161106)
## This PR contains:
- Acceptance test visitor can choose articles by categories.
- Refactored react router.
- Categories.
- Adapted request to fetch categories.

### CODE REVIEW

**Review by mauroavellaneda — APPROVED** [review]

File: src/components/DisplayArticles.jsx [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
   useEffect(() => {
     const getArticlesIndex = async () => {
-      setArticles(await Articles.index());
+      setArticles(await Articles.index(category));
+      
     };
+    
     getArticlesIndex();
-  }, []);
+  });
```
Comment (oliverochman) [inline src/components/DisplayArticles.jsx]: I would listen to changes on the `category` variable here

File: src/components/DisplayArticles.jsx [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
 
   return (
     <div className="articles-container">
+       {/* <h1>{category}</h1> */}
```
Comment (oliverochman) [inline src/components/DisplayArticles.jsx]: commented out code

File: src/modules/articles.js [hunk 3]
```diff
@@ -1,10 +1,16 @@
 import axios from "axios";
 
 const Articles = {
-  async index() {
-    let result = await axios.get("/articles");
-    return result.data.articles;
+  async index(category) {
+    let params = category && { category: category };
+    try {
+      let result = await axios.get("/articles/", { params: params });
```
Comment (oliverochman) [inline src/modules/articles.js]: if you pass in the category param in the URL, you can actually stub out the different categories in cypress

```
let result
if (category) { 
  result = await axios.get(`/articles/?category=${category}`)
} else {
  result = await axios.get("/articles")
}
```

File: cypress/integration/visitorCanChooseArticlesByCategories.feature.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      cy.get("[data-cy='news']").should("contain", "News");
+      cy.get("[data-cy='sports']").should("contain", "Sports");
+      cy.get("[data-cy='politics']").should("contain", "Politics");
+    });
+  });
+
+  context("successfully - News", () => {
+    beforeEach(() => {
+      cy.server();
+      cy.route({
```
Comment (oliverochman) [inline cypress/integration/visitorCanChooseArticlesByCategories.feature.js]: I think this is a false positive, this should not work with your current implementation

File: src/components/Header.jsx [hunk 5]
```diff
@@ -1,11 +1,21 @@
 import React from "react";
 import { Menu } from "semantic-ui-react";
+import { Link} from "react-router-dom";
 
 const Header = () => {
   return (
     <Menu inverted>
       <Menu.Item>
         <h1 data-cy="header">El Gaucho Nyheter</h1>
```
Comment (oliverochman) [inline src/components/Header.jsx:9]: perhaps add a home button that leads to the root_path "/"

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by emtalen — APPROVED** [review]

**Pull request merged**